src/modules-load/modules-load.c

```diff
@@ -17,12 +17,30 @@
 #include "proc-cmdline.h"
 #include "string-util.h"
 #include "strv.h"
+#include <pthread.h>
 
 static char **arg_proc_cmdline_modules = NULL;
 static const char conf_file_dirs[] = CONF_PATHS_NULSTR("modules-load.d");
+static const int max_tasks = 4;
+
+typedef struct {
+        struct kmod_ctx *ctx;
+        char *line;
+        int *result;
+} thread_data;
 
 STATIC_DESTRUCTOR_REGISTER(arg_proc_cmdline_modules, strv_freep);
 
+static void *thread_func(void *arg) {
```

Comment:
We nowadays usually use `void* func()` rather than `void *func()`. That is, please swap a space and `*`.